src/kernel/scheduling/thread.c

```diff
@@ -57,7 +57,7 @@ thread_t* create_thread(char* thread_name, void (*_entry_point)(void *), void* a
     }
 
     // Every thread need it's kernel stack allocated (aka rsp0 field of the TSS)
-    new_thread->rsp0 = kmalloc(THREAD_DEFAULT_STACK_SIZE);
+    new_thread->rsp0 = kmalloc(THREAD_DEFAULT_STACK_SIZE) + THREAD_DEFAULT_STACK_SIZE;
```

Comment:
might be useful to allocate stacks from the vmm, and then you can pass in a flag like VM_STACK - which can automatically return the top address for you (to avoid future errors like this one).

Reply:
this is a cool idea! 
i'll implement it!